packages/material-ui/src/internal/SwitchBase.test.js

```diff
@@ -259,57 +259,71 @@ describe('<SwitchBase />', () => {
   describe('with FormControl', () => {
     describe('enabled', () => {
       it('should not have the disabled class', () => {
-        const { getByTestId } = render(
+        const { getByRole, getByTestId } = render(
           <FormControl>
-            <SwitchBase data-testid="root" icon="unchecked" checkedIcon="checked" type="checkbox" />
+            <SwitchBase
+              data-testid="IconButton"
+              icon="unchecked"
+              checkedIcon="checked"
+              type="checkbox"
+            />
           </FormControl>,
         );
 
-        expect(getByTestId('root')).not.to.have.class(classes.disabled);
+        expect(getByTestId('IconButton')).not.to.have.class(classes.disabled);
+        expect(getByRole('checkbox')).not.to.have.attribute('disabled');
       });
 
       it('should be overridden by props', () => {
-        const { getByTestId } = render(
+        const { getByRole, getByTestId } = render(
           <FormControl>
             <SwitchBase
               disabled
-              data-testid="root"
+              data-testid="IconButton"
               icon="unchecked"
               checkedIcon="checked"
               type="checkbox"
             />
           </FormControl>,
         );
 
-        expect(getByTestId('root')).to.have.class(classes.disabled);
+        expect(getByTestId('IconButton')).to.have.class(classes.disabled);
+        expect(getByRole('checkbox')).to.have.attribute('disabled');
       });
     });
 
     describe('disabled', () => {
       it('should have the disabled class', () => {
-        const { getByTestId } = render(
+        const { getByRole, getByTestId } = render(
           <FormControl disabled>
-            <SwitchBase data-testid="root" icon="unchecked" checkedIcon="checked" type="checkbox" />
+            <SwitchBase
+              data-testid="IconButton"
```

Comment:
Looks like c&p error. The old props seemed fine. In this test we don't care about the underlying component.

Reply:
> In this test we don't care about the underlying component.

Gotcha! Made with a purpose to distinguish tests for class name and attribute. Will rollback